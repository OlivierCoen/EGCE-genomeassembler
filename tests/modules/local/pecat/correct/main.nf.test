nextflow_process {

    name "Test Process PECAT_CORRECT"
    script "modules/local/pecat/correct/main.nf"
    process "PECAT_CORRECT"
    tag "pecat_correct"

    test("Should run without failures") {

        when {

            process {
                """
                input[0] = Channel.of(
                    [
                        [ id: "test", genome_size: 1500000 ],
                        file( '$projectDir/tests/test_data/pecat/input/reads.small.fasta.gz', checkIfExists: true)
                    ]
                )
                input[1] = Channel.fromPath( "${projectDir}/tests/test_data/pecat/input/pecat_optional_config.txt", checkIfExists: true )
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
