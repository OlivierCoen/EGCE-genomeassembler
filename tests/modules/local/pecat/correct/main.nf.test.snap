{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "genome_size": 1500000
                        },
                        "results.tar.gz:md5,b5f413d02b5487b53721fc8a568e83de"
                    ]
                ],
                "1": [
                    [
                        "PECAT_CORRECT",
                        "pecat",
                        "0.0.3"
                    ]
                ],
                "results": [
                    [
                        {
                            "id": "test",
                            "genome_size": 1500000
                        },
                        "results.tar.gz:md5,b5f413d02b5487b53721fc8a568e83de"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-27T16:22:09.429644534"
    }
}